src/legacy/status_im/events.cljs

```diff
@@ -122,7 +124,9 @@
               #(when-let [chat-id (:current-chat-id db)]
                  {:dispatch [:chat/mark-all-as-read chat-id]})
               #(when requires-bio-auth
-                 {:dispatch [:biometric/authenticate {:on-fail on-biometric-auth-fail}]}))))
+                 {:dispatch [:biometric/authenticate {:on-fail on-biometric-auth-fail}]})
+              #(when notifications-settings?
+                 {:dispatch [:notifications/check-notifications-blocked]}))))
```

Comment:
Here we're conditionally dispatching the `:notifications/check-notifications-blocked` event to refresh the screen UI if the user has recently unblocked or allowed notification permissions from outside the Status app and then returned to the notifications settings screen.